target/linux/ath79/dts/qca9561_tplink_archer-c25-v1.dts

```diff
@@ -0,0 +1,218 @@
+// SPDX-License-Identifier: GPL-2.0-or-later OR MIT
+/dts-v1/;
+
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/input/input.h>
+
+#include "qca956x.dtsi"
+
+/ {
+	compatible = "tplink,archer-c25-v1", "qca,qca9560";
+	model = "TP-Link Archer C25 v1";
+
+	chosen {
+		bootargs = "console=ttyS0,115200n8";
+	};
+
+	led_spi {
+		compatible = "spi-gpio";
+		#address-cells = <1>;
+		#size-cells = <0>;
+
+		gpio-sck = <&gpio 15 GPIO_ACTIVE_HIGH>;
+		gpio-mosi = <&gpio 14 GPIO_ACTIVE_HIGH>;
+		cs-gpios = <&gpio 16 GPIO_ACTIVE_HIGH>;
+		num-chipselects = <1>;
+
+		led_gpio: led_gpio@0 {
+			compatible = "fairchild,74hc595";
+			reg = <0>;
+			gpio-controller;
+			#gpio-cells = <2>;
+			registers-number = <1>;
+			spi-max-frequency = <10000000>;
+		};
+	};
+
+	keys {
+		compatible = "gpio-keys-polled";
+		poll-interval = <100>;
+
+		wifi_button {
+			label = "WiFi button";
+			linux,code = <KEY_RFKILL>;
+			gpios = <&gpio 22 GPIO_ACTIVE_LOW>;
+		};
+
+		reset_button {
+			label = "Reset button";
+			linux,code = <KEY_RESTART>;
+			gpios = <&gpio 1 GPIO_ACTIVE_LOW>;
+		};
+	};
+
+	gpio-export {
```

Comment:
Please convert this to use the userspace gpio-switch since gpio-export is never going to get accepted upstream.

See https://github.com/openwrt/openwrt/pull/1366#issuecomment-424834448

Reply:
I read through the thread. So do you want me to just put
ucidef_add_gpio_switch "led_control" "LED control" "21" "0"
ucidef_add_gpio_switch "led_reset" "LED reset" "19" "1"
into 03_gpio_switches?

Reply:
Maybe a fixed-regulator is more suited for this task? https://elixir.bootlin.com/linux/v5.1/source/Documentation/devicetree/bindings/regulator/fixed-regulator.yaml

Reply:
@blocktrron 
Can't say, that's way beyond my DTS skills.
However, if you provide some code, I would test it (but not necessarily as part of this PR).

Reply:
Hmm; tested with ucidef_add_gpio_switch, and all LEDs are either dead or all-on.
So this seems to be the wrong way.

Reply:
Well, after having corrected the sck-gpios issue, this is working now.